std/internal/windows/bcrypt.d

```diff
@@ -0,0 +1,65 @@
+module std.internal.windows.bcrypt;
+
+version (Windows):
+
+import core.sys.windows.bcrypt : BCryptGenRandom, BCRYPT_USE_SYSTEM_PREFERRED_RNG;
+import core.sys.windows.windef : HMODULE, PUCHAR, ULONG;
+import core.sys.windows.ntdef : NT_SUCCESS;
+
+pragma(lib, "Bcrypt.lib");
```

Comment:
Why is this here if ultimately you'll be using the loaded DLL?

Reply:
That’s an excellent question. And that one I’ve been asking myself as well when I copied it over from `std.internal.windows.advapi32`.
<https://github.com/dlang/phobos/blob/393d8264bb04dafe71226ac288a08157f1953294/std/internal/windows/advapi32.d#L17>
Unfortunately, I don’t have an answer to it.
I figured, if the other module has had the pragma for 13 years now, it wouldn’t hurt copying that detail.

Reply:
That said, I would also be interesting in learning why that original `pragma(lib, "Bcrypt.lib");` didn’t do the job.

Reply:
@ibuclaw On Windows the library file loads the correct system DLL. Loading the DLL directly is strongly discouraged as the underlying ABI is not considered stable. Think of it like calling a Kernel method directly.

Reply:
> @ibuclaw On Windows the library file loads the correct system DLL. Loading the DLL directly is strongly discouraged as the underlying ABI is not considered stable. Think of it like calling a Kernel method directly.

A few things come to mind.

1. Is pragma(lib) compiled into archives/static libraries? This is not the case on Linux, you can only link third party libraries against shared objects. (ie: phobos.dll would have a reference but not phobos.lib).

2. pragma(lib) is ignored when coming from an imported module.

3. There isn't a good mechanism for a D library to tell downstream users what its dependencies are (unless you're building with dub). Writing libs to a custom section can work for individual objects but not static libraries.

Reply:
Thanks for elaborating. Yeah, I figured that something like this would be the case when thinking this through again earlier today. (Especially given that Phobos is built in a two step process where linking happens in a separate invocation from the object file building.)